elastic/kibana
# [Cloud Security] limit setup options for Agentless CSPM

## Summary

Follow up after 
- https://github.com/elastic/kibana/pull/171671

Closes
- https://github.com/elastic/security-team/issues/7969

Includes:
- limiting setup options for agentless to only Direct Access Keys and Temporary Keys
- covering Agentless for edit flow

### How to test

Make sure to have the FF in your `serverless.security.dev.yml` (it's similar to the `kibana.dev.yml` but specifically for Serverless Security Projects) enabled. Also specify some serverless project id, to enable the logic of `isServerlessEnabled`
```
xpack.fleet.enableExperimental: ['agentless']
xpack.cloud.serverless.project_id: 'some_fake_project_id'
``` 

The follow the steps from this comment https://github.com/elastic/security-team/issues/7972#issuecomment-1808096052 to have the Agentless artifacts (agent, policy, output, and fleet server host) locally

After that, you should be able to test the flow. 

### Screencast
[screencast-mail.google.com-2023.12.08-16_37_35.webm](https://github.com/elastic/kibana/assets/478762/b94b685f-ed37-4e45-9907-bbd95cb8975a)


### Checklist

Delete any items that are not applicable to this PR.
